### DISCUSSION TITLE
Access boundsProg from within RAYGEN in .cu file

### DISCUSSION THREAD

**Discussion by vani-nag** [owl-project/owl#179] [Q&A] [answered]
Is there a way to refit the acceleration structure from within the RAYGEN program? I'd like to be able to change the radius of my spheres and refit the acceleration structure from within the device code and without having to go back to the host side. Please let me know if this is possible. Thanks!

**Comment by ingowald** [comment, marked as answer]
No, unfortunately there's not; you can only refit through a optix API call.

**Reply by ingowald** [reply]
BTW that's not a _OWL_ limitation, it's a OptiX/RTX limitation, same in Vulkan etc too.
